src/k8s-engine/lima.ts

```diff
@@ -1552,7 +1551,45 @@ export default class LimaBackend extends events.EventEmitter implements K8s.Kube
         ]);
         if (config.enabled) {
           this.lastCommandComment = 'Checking k3s images';
-          await this.progressTracker.action(this.lastCommandComment, 100, this.k3sHelper.ensureK3sImages(desiredVersion));
+          try {
+            await this.progressTracker.action(this.lastCommandComment, 100, this.k3sHelper.ensureK3sImages(desiredVersion));
+          } catch (ex:any) {
+            console.log(`Failed to find version ${ desiredVersion.raw }: ${ ex }`, ex);
+            if (await K3sHelper.failureDueToNetworkProblem('github.com')) {
+              try {
+                const newVersion: semver.SemVer = await K3sHelper.selectClosestImage(desiredVersion);
+
+                if (semver.lt(newVersion, desiredVersion)) {
+                  const options: Electron.MessageBoxOptions = {
+                    message:   `Downgrading from ${ desiredVersion.raw } to ${ newVersion.raw } will lose existing Kubernetes workloads. Delete the data?`,
+                    type:      'question',
```

Comment:
FWIW, this would be `warning` normally, because one option leads to data loss. But we're probably not consistent anyway :D

(Also, it technically would have been `exclamation` on Windows, except I see it's now just an alias to `warning`.)